src/strands/models/bedrock.py

```diff
@@ -246,63 +306,132 @@ def stream(self, request: dict[str, Any]) -> Iterable[dict[str, Any]]:
-                                        "redactUserContentMessage": self.config.get(
-                                            "guardrail_redact_input_message", "[User input redacted.]"
-                                        )
-                                    }
-                                }
-                            if self.config.get("guardrail_redact_output", False):
-                                logger.debug("Found blocked output guardrail. Redacting output.")
-                                yield {
-                                    "redactContent": {
-                                        "redactAssistantContentMessage": self.config.get(
-                                            "guardrail_redact_output_message", "[Assistant output redacted.]"
-                                        )
-                                    }
-                                }
+                        guardrail_data = chunk["metadata"]["trace"]["guardrail"]
+                        if self._has_blocked_guardrail(guardrail_data):
+                            yield from self._generate_redaction_events()
+                    yield chunk
+            else:
+                # Non-streaming implementation
+                response = self.client.converse(**request)
+
+                # Convert and yield from the response
+                yield from self._convert_non_streaming_to_streaming(response)
+
+                # Check for guardrail triggers after yielding any events (same as streaming path)
+                if (
+                    "trace" in response
+                    and "guardrail" in response["trace"]
+                    and self._has_blocked_guardrail(response["trace"]["guardrail"])
+                ):
+                    yield from self._generate_redaction_events()
+
+        except ClientError as e:
+            error_message = str(e)
 
-                yield chunk
-        except EventStreamError as e:
-            # Handle throttling that occurs mid-stream?
-            if "ThrottlingException" in str(e) and "ConverseStream" in str(e):
-                raise ModelThrottledException(str(e)) from e
+            # Handle throttling error
+            if "ThrottlingException" in error_message:
+                raise ModelThrottledException(error_message) from e
 
-            if any(overflow_message in str(e) for overflow_message in BEDROCK_CONTEXT_WINDOW_OVERFLOW_MESSAGES):
+            # Handle context window overflow
+            if any(overflow_message in error_message for overflow_message in BEDROCK_CONTEXT_WINDOW_OVERFLOW_MESSAGES):
                 logger.warning("bedrock threw context window overflow error")
                 raise ContextWindowOverflowException(e) from e
+
+            # Otherwise raise the error
             raise e
-        except ClientError as e:
-            # Handle throttling that occurs at the beginning of the call
-            if e.response["Error"]["Code"] == "ThrottlingException":
```

Comment:
If anything, I think I'd rather get rid of the string check above than getting rid of the direct check here.

I preserved the block above (if "ThrottlingException" in str(e) and "ConverseStream" in str(e):) because I couldn't reproduce the case, but this line with the error-code check is the is right way AFAIK to check for throttling at the API level (not sure about streaming)